application/src/main/kotlin/application/TestCommand.kt

```diff
@@ -82,6 +84,48 @@ class TestCommand : Callable<Unit> {
     @Option(names = ["--filter-not-name"], description = ["Run only tests which do not have this value in their name"], defaultValue = "\${env:SPECMATIC_FILTER_NOT_NAME}")
     var filterNotName: String = ""
 
+    @Option(
+        names= ["--filter"],
+        description = [
+           """
+Filter tests matching the specified filtering criteria
+
+You can filter tests based on the following keys:
+- `METHOD`: HTTP methods (e.g., GET, POST)
+- `PATH`: Request paths (e.g., /users, /product)
+- `STATUS-CODE`: HTTP response status codes (e.g., 200, 400)
+- `HEADERS`: Request headers (e.g., Accept, X-Request-ID)
+- `QUERY-PARAM`: Query parameters (e.g., status, productId)
+- `EXAMPLE-NAME`: Example name (e.g., create-product, active-status)
+
+To specify multiple values for the same filter, separate them with commas. 
+For example, to filter by HTTP methods: 
+--filter="METHOD=GET,POST"
+
+You can supply multiple filters as well. 
+For example:
+--filter="METHOD=GET,POST" --filter="PATH=/users"
+           """
+        ],
+        required = false
+    )
+    var filter: List<String> = emptyList()
+
+    @Option(
+        names= ["--filterNot"],
```

Comment:
this should be `--filter-not`, in keeping with `--filter-name` and `--filter-not-name`

Reply:
Addressed.